### PR TITLE
Fix for DIV-6094

### PR DESCRIPTION
**Pull request by Dizrupto**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DIV-6699

### Change description ###

Fix for functionality defined in https://tools.hmcts.net/jira/browse/DIV-6094 to work again

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/CaseField/CaseField-deemed-and-dispensed-nonprod.json
Comment (madjava) [inline definitions/divorce/json/CaseField/CaseField-deemed-and-dispensed-nonprod.json:229]: Ok this is our way out if this quagmire 😜

**Review by madjava — APPROVED** [review]

File: definitions/divorce/json/CaseField/CaseField-deemed-and-dispensed-nonprod.json
Comment (Dizrupto) [reply definitions/divorce/json/CaseField/CaseField-deemed-and-dispensed-nonprod.json:229]: yep

File: definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json
Comment (jvictorcf) [inline definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json:994]: Are you sure you need these three roles to have CRUD rights to this field?

**Review by jvictorcf — APPROVED** [review]
@Dizrupto, I'm happy with it. 
I have suggestion, though: instead of having two fields, why don't we have one field called something like "LastServiceApplication". This field could be a complex type containing the "type" and "granted" fields.

File: definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json
Comment (Dizrupto) [reply definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json:994]: I am not entirely sure but I thought it would be safer to copy the permissions of the existing fields because I remember that we had added the "CRUD" roles because of a bug we had noticed. Just to avoid having anything like this with those fields I thought it it just easier to give the same permissions.
Also, having 1 complex type field could work assuming that showConditions work well with Complex type, so that we can do: ServiceApplication.type = "true" for example.

Comment (jvictorcf) [reply definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json:994]: If we can make sure we only have the necessary CRUD rights, that'd be better. I'm especially worried about the solicitor having it, if not needed.
As to the complex type, I think sub-field reference works (although I'm not sure).
Happy for you to merge this and create a sub-task on your story to go through these comments (if you think they make sense).

**Review by yarpo — APPROVED** [review]

**Comment by yarpo** [comment]
Once you merge it, can you please make sure PR: https://github.com/hmcts/div-ccd-definitions/pull/595

is updated with the latest commit from master?

just:

```
git pull origin master
git push
```

or press button on the PR.

File: definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json
Comment (jvictorcf) [inline definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField-deemed-and-dispensed-nonprod.json:994]: The only thing I'd check before merging is exactly which roles need to have CRUD rights to this field...

**Review by yarpo — CHANGES_REQUESTED** [review]
please don't merge it yet

**Comment by yarpo** [comment]
It's been already merged into https://github.com/hmcts/div-ccd-definitions/pull/594 via https://github.com/hmcts/div-ccd-definitions/pull/600

**Review by yarpo — APPROVED** [review]

**Pull request merged**
